scripts/distribution/macOS-package/build.sh

```diff
@@ -248,7 +248,7 @@ function collectDylibs() {
         local fileToFix=${1:?"Missing file to fix with dylibbundler"};
         cd "$payloadDir/Library/Concordium Node"
         # Paths to search for dylibs are added with the '-s' flag.
-        "$macdylibbundlerDir/dylibbundler" --fix-file "$fileToFix" --bundle-deps --dest-dir "./libs" --install-path "@executable_path/libs/" --overwrite-dir \
+        "$macdylibbundlerDir/dylibbundler" --fix-file "$fileToFix" --bundle-deps --dest-dir "./libs" --install-path "@executable_path/libs/" --create-dir \
```

Comment:
It is a bit unclear why this fixes it. It would be good to document why we cannot delete the contents of the directory.

As I read it --overwrite-dir implies --create-dir, so it is the deleting part that was causing problems.